### PR TITLE
Improve search

### LINKED ISSUE #6201: Files and folders of disabled spaces appear in search
## Steps to reproduce

1. Create a space ("Testspace"), a folder in that space ("Testfolder") and a file in that folder ("Testfile")
2. Go to global search bar on top, search for "Test"
3. Verify, that the created resources are shown as search results
4. Disable space
5. Search again

## Expected behavior
Files and folders of disabled spaces are not presented as search results.

## Actual behavior
Files and folders of disabled space are presented as search results, clicking on them show "Resource not found"

grafik

### PR DESCRIPTION
**Pull request by aduffeck**
This PR improves and fixes the search service:

1. We no longer return results from disabled spaces (fixes #6201)
2. We fixed a performance regression with big indexes caused by prefix queries which got lost during the experimental-> master merge
3. The sub-searches in the different spaces are now done concurrently

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiContract-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/22259/50/1) failed. Further test are cancelled...

File: services/search/pkg/search/service.go [hunk 1]
```diff
@@ -80,8 +83,17 @@ func (s *Service) Search(ctx context.Context, req *searchsvc.SearchRequest) (*se
 		return nil, err
 	}
 
-	mountpointMap := map[string]string{}
+	spaces := []*provider.StorageSpace{}
 	for _, space := range listSpacesRes.StorageSpaces {
+		if utils.ReadPlainFromOpaque(space.Opaque, "trashed") == "trashed" {
```
Comment (fschade) [inline services/search/pkg/search/service.go:88]: maybe `const _opaqueTrashed =  "trashed"`

File: services/search/pkg/search/service.go [hunk 2]
```diff
@@ -96,94 +108,70 @@ func (s *Service) Search(ctx context.Context, req *searchsvc.SearchRequest) (*se
...
-		res, err := s.engine.Search(ctx, &searchsvc.SearchIndexRequest{
-			Query: req.Query,
-			Ref: &searchmsg.Reference{
-				ResourceId: searchRootID,
-				Path:       mountpointPrefix,
-			},
-			PageSize: req.PageSize,
+	// Spawn workers that'll concurrently work the queue
+	numWorkers := 20
+	if len(spaces) < numWorkers {
```
Comment (fschade) [inline services/search/pkg/search/service.go:131]: can we have `numWorkers` configurable?

**Review by fschade — APPROVED** [review]
just two tiny findings, totally optional. LGTM

File: services/search/pkg/search/service.go (hunk 2 above)
Comment (aduffeck) [reply services/search/pkg/search/service.go:131]: We already have another case where a fixed number of workers is used. I would suggest we leave it at that for now and make it configurable if and when the need arises.

**Pull request merged**